### ISSUE TITLE
PyOpenNI cannot create generators

### ISSUE THREAD

**Issue by larrylisky** [jmendeth/PyOpenNI#63]
Hi, 

I am able to build PyOpenNI and generates But when I run it I keep getting these errors:

$ ./gestures.py 
Initializing...
OpenNI raised an error: Can't create any node of the requested type!

$ ./depth.py 
Traceback (most recent call last):
  File "./depth.py", line 20, in <module>
    depth.create(ctx)
<class 'openni.OpenNIError'>: Can't create any node of the requested type!

It looks like there is a general problem.  

The helloworld.py gives this:

$ ./helloworld.py 
OpenNI version: 1.5.8.5
PyOpenNI version: 2.1.0 alpha

My cmake looks like this:

-- The C compiler identification is GNU 4.8.4
-- The CXX compiler identification is GNU 4.8.4
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Found PythonInterp: /usr/bin/python2.7 (found suitable version "2.7.6", minimum required is "2.7") 
-- Found PythonLibs: /usr/lib/x86_64-linux-gnu/libpython2.7.so (found suitable version "2.7.6", minimum required is "2.7") 
-- Found PkgConfig: /usr/bin/pkg-config (found version "0.26") 
-- Found OpenNI: /usr/lib/libOpenNI.so  
-- OpenNI found (include: /usr/include/ni, lib: /usr/lib/libOpenNI.so)
-- Boost version: 1.54.0
-- Found the following Boost libraries:
--   python
-- Detected CXX compiler flags for Python
-- Configuring done
-- Generating done

Then I copy the openni.so to:

   $ sudo cp -fR lib/openni.so /usr/lib/pymodules/python2.7

**Comment by jmendeth** [comment]
It looks like you don't have any sensor drivers installed for OpenNI to use. Without those drivers, OpenNI cannot communicate with your sensors.

Remember that OpenNI in general was abandoned over a year ago, and this module is deprecated. I have added a notice to the README.

**Comment by larrylisky** [comment]
I found the issue was missing NITE.

**Comment by jmendeth** [comment]
Really? That's strange, NITE isn't needed for the depth demo, AFAIK...

**Comment by larrylisky** [comment]
Agreed, depth generator shouldn't require it, but somehow the dependency is there.  I think you had another user reported similar issue that ended up resolved by installing SimpleOpenNI.

**Comment by jmendeth** [comment]
What sensor do you have?

**Comment by imanakas71** [comment]
I have same problem under the condition below:
OS:Ububntu 14.04 
Sensor: ORBBEC astro pro
OpenNI and OpenNI2 libraries are installed and SimpleView (OpenNI2 C++ sample) is working fine.
Those OpenNI and OpenNI2 were downloaded from ORBBEC 

Driver directory:
/usr/lib/OpenNI2/Drivers:
/usr/lib/OpenNI2/Drivers/OniFile.ini
/usr/lib/OpenNI2/Drivers/PS1080.ini
/usr/lib/OpenNI2/Drivers/PSLink.ini
/usr/lib/OpenNI2/Drivers/libDummyDevice.so
/usr/lib/OpenNI2/Drivers/libDummyDevice.so.0
/usr/lib/OpenNI2/Drivers/libOniFile.so
/usr/lib/OpenNI2/Drivers/libOniFile.so.0
/usr/lib/OpenNI2/Drivers/libPS1080.so
/usr/lib/OpenNI2/Drivers/libPS1080.so.0
/usr/lib/OpenNI2/Drivers/libPSLink.so
/usr/lib/OpenNI2/Drivers/libPSLink.so.0
/usr/lib/OpenNI2/Drivers/liborbbec.so
/usr/lib/OpenNI2/Drivers/orbbec.ini

Error message:

~$ ./depth.py 
Traceback (most recent call last):
  File "./depth.py", line 20, in <module>
    depth.create(ctx)
<class 'openni.OpenNIError'>: Can't create any node of the requested type!

Any suggestion is welcome.
